[Merged by Bors] - Add content-type header to metrics server response

kayibal opened the pull request:
This fixes issues with certain metrics scrapers, which might error if the content-type is not correctly set.

## Issue Addressed

Fixes https://github.com/sigp/lighthouse/issues/3437

## Proposed Changes

Simply set header: `Content-Type: text/plain` on metrics server response. Seems like the errored branch does this correctly already.

## Additional Info

This is needed also to enable influx-db metric scraping which work very nicely with Geth.

kayibal committed:
Add content-type header to metrics server response

This fixes issues with certain metrics scrapers, which might error if the content-type header is not set.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/sigp/lighthouse?pullRequest=3970) <br/>All committers have signed the CLA.

michaelsproul reviewed (changes requested):
Awesome, thanks!

Do you mind applying the same fix to the validator client metrics here?

https://github.com/sigp/lighthouse/blob/38514c07f222ff7783834c48cf5c0a6ee7f346d0/validator_client/src/http_metrics/mod.rs#L124

Thank you! :pray:

kayibal committed:
Set Content-Type header on validator client metrics

Adds compatibility with metric scrapers that fail if no content type header is set.

kayibal commented:
Thanks for the review @michaelsproul ! Sure thing I added it, couldn't find any other place for metrics apart than those two. Also didn't find any tests for the latter.

michaelsproul reviewed (approved):
Looks good, thanks!

michaelsproul commented:
bors r+

michaelsproul commented:
bors r-
bors r+

michaelsproul commented:
bors r-
bors r+
